compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -1179,62 +1179,6 @@ mod test {
         assert_eq!(main.dfg[b1].instructions().len(), 0);
     }
 
-    #[test]
-    fn deduplicate_across_non_dominated_blocks() {
-        let src = "
-            brillig(inline) fn main f0 {
-              b0(v0: u32):
```

Comment:
Can you please re-add this test so that we still test for this behavior? We can just replace the `v4 = add` hoist with an instruction without side effects like `lt`